src/Model/Photo.php

```diff
@@ -21,9 +23,162 @@
+	{
+		if (empty($fields)) {
+			$fields = self::getFields();
+		}
+
+		return DBA::selectFirst("photo", $fields, $condition, $params);
+	}
+
+	/**
+	 * @brief Get a single photo given resource id and scale
+	 *
+	 * This method checks for permissions. Returns associative array
+	 * on success, "no sign" image info, if user has no permission,
+	 * false if photo does not exists
+	 *
+	 * @param string  $resourceid  Rescource ID for the photo
+	 * @param integer $scale       Scale of the photo. Defaults to 0
+	 *
+	 * @return boolean|array
+	 */
+	public static function getPhoto($resourceid, $scale = 0)
+	{
+		$r = self::selectFirst(["uid"], ["resource-id" => $resourceid]);
+		if ($r === false) {
+			return false;
+		}
+
+		$sql_acl = Security::getPermissionsSQLByUserId($r["uid"]);
+
+		$condition = [
+			"`resource-id` = ? AND `scale` <= ? " . $sql_acl,
+			$resourceid, $scale
+		];
+
+		$params = [ "order" => ["scale" => true]];
+
+		$photo = self::selectFirst([], $condition, $params);
+		if ($photo === false) {
+			return self::createPhotoForSystemResource("images/nosign.jpg");
+		}
+		return $photo;
+	}
+
+	/**
+	 * @brief Check if photo with given resource id exists
+	 *
+	 * @param string  $resourceid  Resource ID of the photo
+	 *
+	 * @return boolean
+	 */
+	public static function exists($resourceid)
+	{
+		return DBA::count("photo", ["resource-id" => $resourceid]) > 0;
+	}
+
+	/**
+	 * @brief Get Image object for given row id. null if row id does not exist
+	 *
+	 * @param integer  $id  Row id
+	 *
+	 * @return \Friendica\Object\Image
+	 */
+	public static function getImageForPhoto($photo)
+	{
+		$data = "";
+		if ($photo["backend-class"] == "") {
+			// legacy data storage in "data" column
+			$i = self::selectFirst(["data"], ["id"=>$photo["id"]]);
+			if ($i === false) {
+				return null;
+			}
+			$data = $i["data"];
+		} else {
+			$backendClass = $photo["backend-class"];
+			$backendRef = $photo["backend-ref"];
+			$data = $backendClass::get($backendRef);
+		}
+
+		if ($data === "") {
+			return null;
+		}
+		return new Image($data, $photo["type"]);
+	}
+
+	/**
+	 * @brief Return a list of fields that are associated with the photo table
+	 *
+	 * @return array field list
+	 */
+	private static function getFields()
+	{
+		$allfields = DBStructure::definition(false);
+		$fields = array_keys($allfields["photo"]["fields"]);
+		array_splice($fields, array_search("data", $fields), 1);
+		return $fields;
+	}
+
+	/**
+	 * @brief Construct a photo array for a system resource image
+	 *
+	 * @param string  $filename  Image file name relative to code root
+	 * @param string  $mimetype  Image mime type. Defaults to "image/jpeg"
+	 *
+	 * @return array
+	 */
+	public static function createPhotoForSystemResource($filename, $mimetype = "image/jpeg")
+	{
+		$fields = self::getFields();
+		$values = array_fill(0, count($fields), "");
+		$photo = array_combine($fields, $values);
+		$photo["backend-class"] = "\Friendica\Model\Storage\SystemResource";
```

Comment:
When using double quotes, it's safer to escape backslashes as they are used to denote special characters. This is one of the reasons I prefer single quotes as a default. Please change to `"\\Friendica\\Model\\Storage\\SystemResource"`.